.github/workflows/build_binaries.yml

```diff
@@ -3,33 +3,63 @@
 # separate terms of service, privacy policy, and support
 # documentation.
 
+# If branch, build name will be the branch name and the date of the day 
+# Else if tag, build name will be tag name
+
 name: Build Binaries 
 
 on:
   push:
     branches: [ "main" ]
+    tags:
+      - '*'
+
   pull_request:
     branches: [ "main" ]
 
 jobs:
-  build:
+
+  build_apk:
     runs-on: ubuntu-latest
 
     steps:
+
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::$(echo $GITHUB_REF | cut -d / -f 3)
+
+      - name: Get current date
+        id: date
+        run: echo "today=$(date +'%Y.%m.%d.%H')" >> $GITHUB_ENV
+
       - name: Clone repository
         uses: actions/checkout@v4
       - name: Set up Flutter
         uses: subosito/flutter-action@v2
         with:
           flutter-version: 3.19.0
       - run: flutter pub get
-     # - run: flutter test
-      - run: flutter build apk
-      - run: flutter build appbundle
+      # - run: flutter test
+      # - name: Build linux version
+      #  run: flutter build linux --release
+      - name: build apk with tag 
+        if: startsWith(github.ref, 'refs/tags/')
+        run: flutter build apk --build-name ${{ steps.get_version.outputs.VERSION }}
+
+      - name: build appbundle with tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: flutter build appbundle --build-name ${{ steps.get_version.outputs.VERSION }}
 
+      - name: build apk with branch
+        if: (!startsWith(github.ref, 'refs/tags/'))
+        run: flutter build appbundle --build-name ${{ env.today }}
+
+      - name: build apk with branch
+        if: (!startsWith(github.ref, 'refs/tags/'))
+        run: flutter build appbundle --build-name ${{ env.today }}
+      
```

Comment:
Create a comment to explain which steps are for tags and which are for branches (when I look, I understand but I prefer a clean comment